### PR TITLE
feat(publish-pypi): add support to pyproject.toml python projects and…

### PR DESCRIPTION
**Pull request by acostapazo**
… hatch builder

### CODE REVIEW

File: .github/workflows/publish-pypi.yml [hunk 1]
```diff
@@ -55,12 +54,29 @@ jobs:
...
+      - name: Install, Build and Publish
         run: |
-          python setup.py sdist bdist_wheel
+          if [ -f setup.py ]; then
+              echo "Install, build and publish (setup.py)" >&2
+              pip install setuptools wheel
+              python setup.py sdist bdist_wheel
+          elif [ -f pyproject.toml ]; then
+            echo "Install, build and publish (pyproject.toml)" >&2
+            pip install hatch
```
Comment (fgsalomon) [inline .github/workflows/publish-pypi.yml:65]: What is hatch? Why we need it?

Comment (fgsalomon) [reply .github/workflows/publish-pypi.yml:65]: I only knew about poetry. Poetry seems more popular but it seems there are some discussion about their decisions...

Comment (acostapazo) [reply .github/workflows/publish-pypi.yml:65]: It is quite standard package to build libraries and migrate from setup.py 

Doc: https://hatch.pypa.io/latest/

Examples:
* https://github.com/tiangolo/fastapi/blob/master/pyproject.toml#L1
* https://github.com/pydantic/pydantic/blob/main/pyproject.toml#L2
* https://github.com/encode/starlette/blob/master/pyproject.toml#L2

Comment (acostapazo) [reply .github/workflows/publish-pypi.yml:65]: > "Poetry is mainly used for managing an _application_ and its dependencies whereas Hatch is more agnostic to the project type and offers plugin-based functionality for the entire workflow (versioning, tox-like environments, publishing) so you can easily build things other than wheel/sdist, test in a Docker container, etc ..."

**Pull request merged**